utils/e2e-tests/ts/tests/swap/nativeToEvm.ts

```diff
@@ -0,0 +1,191 @@
+import { expect, it, describe, assert } from "vitest";
+import { RunNodeState, runNode } from "../../lib/node";
+import * as substrate from "../../lib/substrate";
+import { beforeEachWithCleanup } from "../../lib/lifecycle";
+import { Keyring } from "@polkadot/api";
+import { Codec, IEvent } from "@polkadot/types/types";
+import sendAndWait from "../../lib/substrateSendAndAwait";
+import * as eth from "../../lib/ethViem";
+import { getNativeBalance } from "../../lib/substrateUtils";
+
+type EvmSwapBalancesSwappedEvent = Record<
+  "from" | "withdrawedAmount" | "to" | "depositedAmount" | "evmTransactionHash",
+  Codec
+>;
+
+type EthereumExecutedEvent = Record<
+  "from" | "to" | "transactionHash" | "exitReason",
+  Codec
+>;
+
+type TransactionPaymentEvent = Record<"who" | "actualFee", Codec>;
+
+const bridgePotEvmAddress = "0x6d6f646c686d63732f656e310000000000000000";
+const bridgePotNativeAccount =
+  "hmpwhPbL5XJM1pYFVL6wRPkUP5gHQyvC6R5jMkziwnGTQ6hFr";
+const feesPotNativeAccount =
+  "hmpwhPbL5XJTYPWXPMkacfqGhJ3eoQRPLKphajpvcot5Q5zkk";
+
+describe("native to evm tokens swap", () => {
+  let node: RunNodeState;
+  let substrateApi: substrate.Api;
+  let ethPublicClient: eth.PublicClientWebSocket;
+  beforeEachWithCleanup(async (cleanup) => {
+    node = runNode({ args: ["--dev", "--tmp"] }, cleanup.push);
+
+    await node.waitForBoot;
+
+    substrateApi = await substrate.apiFromNodeWebSocket(node, cleanup.push);
+    ethPublicClient = eth.publicClientFromNodeWebSocket(node, cleanup.push);
+  }, 60 * 1000);
+
+  it("success", async () => {
```

Comment:
I don't see where we test the visibility on the EVM side; should be a check where we use `ethPublicClient` to get the block and check that there is a tx with the expected txhash in it and expected `to`, `from`, `value` and `data` (`0x`). This is what we're after.

Reply:
There is a check on ethereum execution event containing the expected evm transaction hash.

Btw, thanks for noting, will add more details including mentioned fields

Reply:
> There is a check on ethereum execution event containing the expected evm transaction hash.

Yes, but that's not really where all those systems that work with EVM RPCs look at. So that's good, but not really relevant for what we're trying to solve here - and there are a lot of ways this can go wrong even if the event at substrate side exists.

Reply:
https://github.com/humanode-network/humanode/pull/1449/commits/45086a6358479c99af1a94ecaaa7d154ffbb287a